osu.Game/Overlays/Direct/DirectListPanel.cs

```diff
@@ -128,7 +157,28 @@ private void load(LocalisationEngine localisation)
                         },
                     },
                 },
+                progressBar = new Box
+                {
+                    Anchor = Anchor.BottomLeft,
+                    Origin = Anchor.BottomLeft,
+                    RelativeSizeAxes = Axes.X,
+                    BypassAutoSizeAxes = Axes.Y,
+                    Size = new Vector2(0, 3),
+                    Alpha = 0,
+                    Colour = colours.Yellow,
+                },
             });
+
+            PreviewPlaying.ValueChanged += newValue => playButton.FadeTo(newValue || IsHovered ? 1 : 0, 120, Easing.InOutQuint);
+            PreviewPlaying.ValueChanged += newValue => progressBar.FadeTo(newValue ? 1 : 0, 120, Easing.InOutQuint);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (PreviewPlaying && playButton.Track != null)
+                progressBar.Width = (float)(playButton.Track.CurrentTime / playButton.Track.Length);

```

Comment:
this is really ugly. at this point the track should be stored locally, not inside the button.